### ISSUE TITLE
Question about generating conditional input embedding

### ISSUE THREAD

**Issue by yuzhang0403** [naver-ai/dap-cl#8]
Hello, thank you for your amazing work and codes. I've noted that the generation of the conditional input embedding ('eimagei') may not match the CIL assumptions. The subsequent code (lines 208~223 in /src/models/vit_dap/vit.py) employs the major prompt from all the samples in the batch to generate the "eimagei", but may it be plausible to presume that all the samples in the batch stem from the same task?
`

        
        dap_prompt_key_norm = F.normalize(self.dap_key_embeddings, dim=-1)

        x_embed_norm = F.normalize(x_cls_embed, dim=-1)
        sim = torch.matmul(dap_prompt_key_norm,
                           torch.transpose(x_embed_norm, 1, 0))

        sim = torch.transpose(sim, 1, 0)
        (sim_top_k, idx) = torch.topk(sim, self.top_k)
        idx = idx.squeeze(dim=-1)

        prompt_id, id_counts = torch.unique(idx, return_counts=True)
        _, major_idx = torch.topk(id_counts, self.top_k)
        major_prompt_id = prompt_id[major_idx]
        idx = expand_to_batch(major_prompt_id, x_cls_embed.shape[0]).squeeze(dim=-1)

        task_id = major_prompt_id.cpu()[0]
`
Put it simply, assuming that all samples in a batch are from the same mission, is this a known condition for CIL?

**Comment by Programmergg** [comment]
I agree with your point of view very much. This stipulation that a batch belongs to the same task will seriously affect test performance. The authors input tasks into the network one by one during testing, which will make it easier for the network to assign a task id to these samples, greatly reducing the difficulty of classification. What if ten tasks are input into the network at the same time? The results of the authors' approach will yield severe degradation.

**Comment by Programmergg** [comment]
The authors followed the same process of assigning task ids during testing as during training, which resulted in erroneous test results.
